src/server/api/routers/group.ts

```diff
@@ -1,35 +1,59 @@
 import { z } from "zod";
+import { TRPCError } from '@trpc/server';
 
 import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";
+import { randomPassphrase } from '~/server/lib/randomPassphrase';
+import { shuffle } from '~/server/lib/shuffle';
+import type { Draw } from '~/server/lib/shuffle';
+
+interface DrawEnhanced {
+  giver: string;
+  receiver: string;
+  passphrase: string;
+}
 
 export const groupRouter = createTRPCRouter({
   postGroup: publicProcedure
```

Comment:
Maybe you can define a return type here like:
https://stackoverflow.com/questions/13315131/enforcing-the-type-of-the-indexed-members-of-a-typescript-object